src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -0,0 +1,594 @@
+        }
+
+        public TimeSpan ConnectTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property ConnectTimeout is not supported.");
+            set => throw new PlatformNotSupportedException("Property ConnectTimeout is not supported.");
+        }
+
+        public TimeSpan Expect100ContinueTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property Expect100ContinueTimeout is not supported.");
+            set => throw new PlatformNotSupportedException("Property Expect100ContinueTimeout is not supported.");
+        }
+
+        public IDictionary<string, object?> Properties => throw new PlatformNotSupportedException("Property Properties is not supported.");
+        protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            return DoFetch(request, cancellationToken);
+        }
+
+        private async Task<HttpResponseMessage> DoFetch(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+
+            var tcs = new TaskCompletionSource<HttpResponseMessage>();
+
+            // Wrap the cancellationToken in a using so that it can be disposed of whether
+            // we successfully fetched or failed trying.
+            using (cancellationToken.Register(() => tcs.TrySetCanceled()))
+            {
+                try
+                {
+                    var requestObject = new JSObject();
+
+                    if (request.Properties.TryGetValue("WebAssemblyFetchOptions", out object? fetchOoptionsValue) &&
+                        fetchOoptionsValue is IDictionary<string, object> fetchOptions)
+                    {
+                        foreach (KeyValuePair<string, object> item in fetchOptions)
+                        {
+                            requestObject.SetObjectProperty(item.Key, item.Value);
+                        }
+                    }
+
+                    requestObject.SetObjectProperty("method", request.Method.Method);
+
+                    // We need to check for body content
+                    if (request.Content != null)
+                    {
+                        if (request.Content is StringContent)
+                        {
+                            requestObject.SetObjectProperty("body", await request.Content.ReadAsStringAsync().ConfigureAwait(continueOnCapturedContext: true));
+                        }
+                        else
+                        {
+                            // 2.1.801 seems to have a problem with the line
+                            // using (var uint8Buffer = Uint8Array.From(await request.Content.ReadAsByteArrayAsync ()))
+                            // so we split it up into two lines.
+                            var byteAsync = await request.Content.ReadAsByteArrayAsync().ConfigureAwait(continueOnCapturedContext: true);
```

Comment:
There's no way to stream the content via the browser APIs?  This could end up resulting in buffering huge amounts of data, depending on the source content.  Maybe it's not relevant for the browser scenarios?

Reply:
Yes there is streaming and is controlled by setting a property per request as was discussed with blazor and asp.net in a recent PR https://github.com/mono/mono/pull/19260.  

gRPC and ASP.NET Core SignalR use the streaming functionality provided by wasm.

Reply:
I'm not understanding then. The code you have here isn't streaming the data... It's buffering the entire request payload into a byte array, which it won't send until it's read to EOF.

Are you saying this is temporary and it'll be revised subsequently?

Reply:
If the browser context supports streaming then streaming can be selected:  `new StreamContent(wasmHttpReadStream = new WasmHttpReadStream(status)`

```
httpResponse.Content = StreamingSupported && streamingEnabled
                        ? new StreamContent(wasmHttpReadStream = new WasmHttpReadStream(status))
                        : (HttpContent)new BrowserHttpContent(status);
```

The browser `fetch` API can programmatically read and manipulate streams of data received over the network.

Reply:
That's about the response content.  I'm talking about the request content. The code at https://github.com/dotnet/runtime/pull/35573/files#diff-4353591640f3202f481b9b246f56a41fR150-R164 is buffering the entire request content into memory.  As written, it can't be streamed.

Reply:
An issue has been opened [here]{https://github.com/dotnet/runtime/issues/36634) as we need more information.  This does not seem to be reliably implemented on the browser side right now.  See [issue](https://github.com/dotnet/runtime/issues/36634) that was opened to continue discussion and address your review comments.

Stephen If this is not sufficient please let me know the correct channels to continue with this.  We should probably continue our discussion on the issue that was opened.